cores/cosa/Cosa/INET/SNMP.cpp

```diff
@@ -114,11 +115,14 @@ int
 SNMP::OID::match(const uint8_t* coid, bool flag)
 {
   uint8_t clen = pgm_read_byte(&coid[0]);
-  if (length < clen) return (-1);
-  for (uint8_t i = 0; i < clen; i++) 
-    if (name[i] != pgm_read_byte(&coid[i + 1])) return (-1);
+  for (uint8_t i = 0; i < clen; i++) {
+    if (i >= length) return (-1);
```

Comment:
It is inside the loop now so as to be able to catch the -2 condition, see next comment.